File: java/v1/src/main/java/com/api/v1/services/customer/CustomerRegistrationService.java
package com.api.v1.services.customer;

import com.api.v1.dtos.requests.CustomerRegistrationRequestDto;
import com.api.v1.dtos.responses.CustomerResponseDto;

import reactor.core.publisher.Mono;

public interface CustomerRegistrationService {

    Mono<CustomerResponseDto> register(CustomerRegistrationRequestDto request);
    
}

File: kotlin/v2/src/main/kotlin/com/api/v2/customer/service/CustomerRegistrationService.kt
package com.api.v2.customer.service

import com.api.v2.customer.domain.Customer
import com.api.v2.customer.dtos.CustomerModificationRequestDto
import com.api.v2.customer.dtos.CustomerRegistrationRequestDto
import com.api.v2.customer.dtos.CustomerResponseDto

interface CustomerRegistrationService {

    suspend fun register(requestDto: CustomerRegistrationRequestDto): CustomerResponseDto
    suspend fun register(existingCustomer : Customer, requestDto: CustomerModificationRequestDto): CustomerResponseDto

}